File: fullstack-online/unit_07/lectures/post_it/server/src/models/Album.js
import { Schema } from "mongoose";



export const AlbumSchema = new Schema({
  title: { type: String, minLength: 3, maxLength: 25, required: true },
  description: { type: String, minLength: 15, maxLength: 250 },
  coverImg: { type: String, required: true, maxLength: 500 },
  archived: { type: Boolean, required: true, default: false },
  category: { type: String, enum: ['aesthetics', 'food', 'games', 'animals', 'vibes', 'misc'], default: 'misc' },

  creatorId: { type: Schema.ObjectId, ref: 'Account', required: true }
}, { timestamps: true, toJSON: { virtuals: true } })

AlbumSchema.virtual('creator', {
  localField: 'creatorId',
  ref: 'Account',
  foreignField: '_id',
  justOne: true
})

File: fullstack-online/unit_09/post_it/server/Models/Album.cs
using System.ComponentModel.DataAnnotations;

namespace post_it.Models;

public class Album
{
  public int Id { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
  [MinLength(3), MaxLength(25)] public string Title { get; set; }
  [MinLength(15), MaxLength(250)] public string Description { get; set; }
  [MaxLength(1000), Url] public string CoverImg { get; set; }
  public bool Archived { get; set; }
  public string Category { get; set; }
  public string CreatorId { get; set; }
  public Profile Creator { get; set; }
}